shared/naturalcrit/markdown.js

```diff
@@ -206,6 +206,38 @@ const mustacheInjectBlock = {
 	}
 };
 
+const superSubScripts = {
+	name  : 'superSubScripts',
+	level : 'inline',
+	start(src) { return src.match(/\^\^[^\s].+[^\s]\^\^/m)?.index; },  // Hint to Marked.js to stop and check for a match
+	tokenizer(src, tokens) {
+		const superRegex = /^\^\^([^\s\^][^\^]*[^\s\^])\^\^/m;
```

Comment:
Unfortunately this approach requires a minimum of 2 characters inside the `^^` because you require a non-space at the start, and another non-space at the end.

After fiddling around on regex101.com, I think the following approach will be the fastest:
```suggestion
		const superRegex = /^\^\^(?!\s)(?=([^\n\^]*[^\s\^]))\1\^\^/m;
```

The `(?!\s)` is a negative lookahead. It will check that there is no space without actually capturing it, which fixes the 2-character minimum.

The `(?=(...))\1` uses a positive lookahead to mimic the behavior of Atomic Groups `(?>...)` which are not available in Javascript. Essentially if gets to the end of the Regex and finds no match, it won't try to backtrack to find a valid end somewhere it missed. Greatly speeds up the `^^whitespace at the end ^^` case.

The `[^\n\^]*[^\s\^]` ensures that there are no newlines or `^` inside. I also added a check to the final character so it can't be another `^`.

Of course this doesn't allow any nesting of superscript or subscript. Doing that in a predictable way is a lot more involved and can't be solved with RegEx alone. Though if we decide to add that capability in a later PR, we can just copy the source from Marked.js and use the same approach they use to handle nested asterisks.

Reply:
> Of course this doesn't allow any nesting of superscript or subscript. Doing that in a predictable way is a lot more involved and can't be solved with RegEx alone. Though if we decide to add that capability in a later PR, we can just copy the source from Marked.js and use the same approach they use to handle nested asterisks.

Is this a thing that happens?